src/install.c

```diff
@@ -748,6 +748,8 @@ static gboolean pre_install_checks(gchar* bundledir, GList *install_images, GHas
 		if (!mfimage->filename && mfimage->hooks.install)
 			goto skip_filename_checks;
 
+		g_assert_nonnull(mfimage->filename);
```

Comment:
That seems like a round about way to say:

```
if (!mfimage->filename)
   goto skip_filename_checks;

g_assert_nonnull(mfimage->hooks.install);

That looks cleaner IMO.

Reply:
Or maybe just check that mfimage->hooks.install is set and raise a GError otherwise?

Reply:
> That seems like a round about way to say:
> 
> ```
> if (!mfimage->filename)
>    goto skip_filename_checks;
> 
> skip_filename_checks:
>   g_assert_nonnull(mfimage->hooks.install);
> ```
> 
> That looks cleaner IMO.

A little cleaner maybe, but not sure if it better describes the intention. In my version, the label that states to skip the checks is used only exactly for that 'special' purpose (when no filename is set but a custom install hook used). In your approach, we jump there and then check the hook like a side note.

> Or maybe just check that mfimage->hooks.install is set and raise a GError otherwise?

At some point, we have to decide if we want GErrors for uncheckable code paths or not.  As I stated in the description of the commit, this cannot be reached by normal code flow as the config file parser already rejects an inappropriate configuration. Thus it is no user error we check here.
Despite I've also had a GError approach in the first round, I decided to only add an assert to satisfy the checker which only handles the function context.

Another option would be to say

``` c
if (!mfimage->filename) {
    if (mfimage->hooks.install)
        goto skip_filename_checks;
    else
        g_assert_not_reached();
}
```

Reply:
Hmm, the last option doesn't look too bad. Is the checker clever enough to understand this?

We should also have a comment for the `g_assert_not_reached()` to point to the location where this precondition is checked.

Reply:
I've updated the PR to match the last example and also added an inline note.
According to my private coverity run, it seems that the original issue is considered resolved now.